src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -59,19 +61,54 @@ void check_sshfs_exists(mp::SSHSession& session)
     run_cmd(session, "which sshfs", error_handler);
 }
 
-void make_target_dir(mp::SSHSession& session, const std::string& target)
+// Return the home directory of the current user.
+std::string get_home_dir(mp::SSHSession& session)
 {
-    run_cmd(session, fmt::format("sudo mkdir -p \"{}\"", target));
+    std::string home = run_cmd(session, "bash -c 'echo ${HOME}'");
+
+    home.pop_back(); // Remove the trailing newline.
+    home += "/";
+
+    return home;
+}
+
+// Given any path, return it expressed as an absolute path.
+std::string get_absolute_path(mp::SSHSession& session, const std::string& path)
+{
+    return *(path.begin()) == '/' ? path : get_home_dir(session) + '/' + path;
+}
+
+// Given a path, returns the part of the path which exists.
+std::string get_existing_dirs(mp::SSHSession& session, const std::string& path)
+{
+    std::string existing = run_cmd(
+        session, fmt::format("/bin/bash -c 'P=\"{}\"; while [ ! -d \"$P/\" ]; do P=${{P%/*}}; done; echo $P/'", path));
+
+    // Strip the trailing newline from the string.
+    existing.pop_back();
+
+    return existing;
+}
+
+// Create a directory on a given root folder.
+void make_target_dir(mp::SSHSession& session, const std::string& root, const std::string& target)
+{
+    if (!target.empty())
+        run_cmd(session, fmt::format("cd \"{}\" && sudo mkdir -p \"{}\"", root, target));
 }
 
-void set_owner_for(mp::SSHSession& session, const std::string& target)
+// Set ownership of all directories on a path starting on a given root.
+// Assume it is already created.
+void set_owner_for(mp::SSHSession& session, const std::string& root, const std::string& target)
 {
     auto vm_user = run_cmd(session, "id -nu");
     auto vm_group = run_cmd(session, "id -ng");
     mp::utils::trim_end(vm_user);
     mp::utils::trim_end(vm_group);
 
-    run_cmd(session, fmt::format("sudo chown {}:{} \"{}\"", vm_user, vm_group, target));
+    run_cmd(session, fmt::format("/bin/bash -c 'cd \"{}\" && P=\"{}\"; Q=$P; while [ ! -z \"$Q\" ]; do sudo chown "
+                                 "{}:{} $P; Q=`echo $P | cut -f 1 -d / -s`; P=${{P%/*}}; done'",
+                                 root, target, vm_user, vm_group));
```

Comment:
This could then just be:
```cpp
    run_cmd(session, fmt::format("/bin/bash -c 'cd \"{}\" && sudo chown -R {}:{} \"{}\"'", root, vm_user, vm_group, relative_target));
```

Reply:
Right, it could be that because we just created the path.